packages/node/src/runtimePlugin.ts

```diff
@@ -0,0 +1,320 @@
+        }
+        console.log('adding remote', chunkId);
+        callback(enhancedRemote);
+      })
+      .catch(function (error) {
+        callback(error);
+      });
+  };
+  // Dynamic filesystem chunk loading for javascript
+  if (__webpack_require__.f) {
+    const handle = function (chunkId, promises) {
+      var installedChunkData = installedChunks[chunkId];
+      if (installedChunkData !== 0) {
+        // 0 means "already installed".
+        if (installedChunkData) {
+          promises.push(installedChunkData[2]);
+        } else {
+          if (__webpack_require__.federation.chunkMatcher(chunkId)) {
+            // check if real chunk for handler
+            var promise = new Promise(function (resolve, reject) {
+              installedChunkData = installedChunks[chunkId] = [resolve, reject];
+
+              function installChunkCallback(error, chunk) {
+                if (error) return reject(error);
+                if (chunk) installChunk(chunk);
+                resolve(chunk);
+              }
+
+              var fs =
+                typeof process !== 'undefined'
+                  ? __non_webpack_require__('fs')
+                  : false;
+              var filename =
+                typeof process !== 'undefined'
+                  ? resolveFile(
+                      __webpack_require__.federation.rootOutputDir || '',
+                      chunkId,
+                    )
+                  : false;
+
+              if (fs && fs.existsSync(filename)) {
+                loadChunkStrategy(
+                  'filesystem',
+                  chunkId,
+                  __webpack_require__.federation.rootOutputDir || '',
+                  undefined,
+                  installChunkCallback,
+                );
+              } else {
+                var chunkName = __webpack_require__.u(chunkId);
+                const loadingStrategy =
+                  typeof process === 'undefined' ? 'http-vm' : 'http-eval';
+                loadChunkStrategy(
+                  loadingStrategy,
+                  chunkName,
+                  __webpack_require__.federation.initOptions.name,
+                  __webpack_require__.federation.initOptions.remotes,
+                  installChunkCallback,
+                );
+              }
+            });
+            promises.push((installedChunkData[2] = promise));
+          } else {
+            installedChunks[chunkId] = 0;
+          }
+        }
+      }
+    };
+    if (__webpack_require__.f.require) {
+      console.warn(
+        '\x1b[33m%s\x1b[0m',
+        'CAUTION: build target is not set to "async-node", attempting to patch additional chunk handlers. This may not work',
+      );
+      __webpack_require__.f.require = handle;
+    }
+    if (__webpack_require__.f.readFileVm) {
+      __webpack_require__.f.readFileVm = handle;
+    }
+  }
+})();
+
+export default function () {
+  return {
+    name: 'node-internal-plugin',
```

Comment:
So you average the iife runtime plugin to inject webpack internal runtime , really nice idea! But can we move the logic to `beforeInit` ? Because the mf runtime plugins just expose name seems a little strange ...

It's not mandatory . And if make the iife logic to beforeInit is too late , it can keep status quo as well .

Reply:
The reason for placing it in the module directly is to patch webpack require as soon as possible, not after federation  has begun to initialize. 

I can try beforeInit as well